### PR TITLE
LUNAR HOTEL BACKEND(COMPLETE PROJECT)

### PR DESCRIPTION
**Pull request by DammyShittu**
## To set up a full project, we completed the following requirements,

- You should follow the layout of the provided design. You should change only the titles, descriptions, and photos - in order to create a website about something other than motorcycles.

- Select a theme for your website - is it going to be a website for booking doctor appointments, booking online classes, or something else?

- The user logs in to the website by typing the email and password.

- In the navigation panel, the user can see links to:

- Motorcycles/doctors/classes/items that you selected as a theme, (in our case, ROOMS)

- "My reservations".

- "Create Room"

- "Log Out"

- On the main page, the user can see a list of motorcycles/doctors/classes/items that you selected as a theme.

- When the user selects a specific item, they can see the details page with its full description (skip the "Rotate image" button).

- In the details page, the user can click the "Reserve" button (in the design you can see the "Configure" button - please replace it with the "Reserve" button).

- When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item.

- Make the app responsive, creating both mobile and desktop versions.

- When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with title and "Delete" button.

- When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.

- To reserve an appointment, the user has to select a date and city (username and selected item are autofilled).

- Use the design based on the "Book a vespa test-ride" and add all necessary inputs.

- When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date, and city).

### CODE REVIEW

File: app/controllers/api/v1/reservations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+class Api::V1::ReservationsController < ApplicationController
+  before_action :set_reservation, only: :destroy
+
+  def index
+    @reservations = current_user.reservations
+  end
+
+  def show
+    render json: @reservation, status: :ok
+  end
```
Comment (MuriloRoque) [inline app/controllers/api/v1/reservations_controller.rb]: - [ ] I believe you are not using this action, maybe it's better to remove it.

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+Rails.application.routes.draw do
+  mount Rswag::Ui::Engine => '/'
+  mount Rswag::Api::Engine => '/api-docs'
+  devise_for :users, defaults: { format: :json }, path: 'users',
+                     path_names: { sign_in: 'login', sign_out: 'logout', registration: 'signup' }
+
+  namespace :api, defaults: { format: :json } do
+    namespace :v1 do
+      resources :rooms, only: %i[index create show destroy]
+      resources :reservations, only: %i[index show create destroy]
```
Comment (MuriloRoque) [inline config/routes.rb]: - [ ] Like I pointed above, you aren't using the `show` action here, so I would remove the route.

**Review by MuriloRoque — CHANGES_REQUESTED** [review]
Hi @DammyShittu, @medaminedev66, @julie-ify, @mmatongo, 

Very good job so far!
There are some very small issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/controllers/api/v1/reservations_controller.rb (hunk 1 above)
Comment (julie-ify) [reply app/controllers/api/v1/reservations_controller.rb]: Yes, you're right. It has been removed!

File: config/routes.rb (hunk 2 above)
Comment (julie-ify) [reply config/routes.rb]: Done!

**Review by MuriloRoque — APPROVED** [review]
Hi @DammyShittu, @julie-ify, @medaminedev66, @mmatongo, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by mmatongo** [comment]
> Hi @DammyShittu, @julie-ify, @medaminedev66, @mmatongo,
> 
> In my opinion, your project is ready for the presentation! There is nothing else to say other than... it's time to merge it :shipit:
> 
> Good luck! 🎉
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


Thank you

**Pull request merged**